src/model/client/AbstractClient.ts

```diff
@@ -161,18 +157,21 @@ export abstract class AbstractClient {
    */
   protected normalizeOptions<E extends AbstractEndpoint>(
     endpoint: E,
-    options: OptionsWithBody<E> | OptionsWithoutBody<E>,
-  ): OptionsWithBody<E> | OptionsWithoutBody<E> {
-    // Merge all headers
-    options.headers = {
-      ...this.getHeaders(),
-      ...options.headers,
-      ...endpoint.getHeaders(),
+    options: Options<E>,
+  ): WithRequired<Options<E>, 'headers'> {
+    const newOptions: WithRequired<OptionsWithBody<E>, 'headers'> = {
+      ...options,
+      headers: {
+        ...(['POST', 'PUT'].includes(endpoint.method.toUpperCase()) ? {'Content-Type': 'application/json'} : {}),
```

Comment:
Hebben we ook Constant values voor de POST & PUT? aangezien we alles typen.

Reply:
nee, in ts vind ik het altijd lastig om daar een lijn in te trekken omdat die types de behoefte aan consts in veel gevallen al wegnemen. Ik kan wel dit arraytje een const maken op zich

Reply:
Nee, dan laten het zo; Ik verwacht dat dit wel goed zal gaan.